src/support/err.c

```diff
@@ -369,23 +372,20 @@ __wt_panic_func(WT_SESSION_IMPL *session, int error, const char *func, int line,
       __eventv(session, false, WT_PANIC, func, line, "the process must exit and restart", ap));
     va_end(ap);
 
-/*
- * Confusing #ifdef structure because gcc/clang knows the abort call won't return, and Visual Studio
- * doesn't.
- */
 #if defined(HAVE_DIAGNOSTIC)
-    __wt_abort(session); /* Drop core if testing. */
-                         /* NOTREACHED */
+    /* Drop core if testing, unless it is data corruption and we have been asked not to panic. */
+    if (!F_ISSET(conn, WT_CONN_DATA_CORRUPTION) ||
+      FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_CORRUPTION_ABORT))
```

Comment:
Minor: I suggest rewriting the comment so it is the same logic as the code.

Reply:
done. I added a more elaborate comment on what we are trying to do.